esphome/core/esphal.cpp

```diff
@@ -29,9 +29,15 @@ GPIOPin::GPIOPin(uint8_t pin, uint8_t mode, bool inverted)
       gpio_mask_(pin < 16 ? (1UL << pin) : 1)
 #endif
 #ifdef ARDUINO_ARCH_ESP32
+#ifdef CONFIG_IDF_TARGET_ESP32C3
+          gpio_set_(pin < 32 ? &GPIO.out_w1ts.val : &GPIO.out_w1tc.val),
```

Comment:
I think this should just be:
```suggestion
          gpio_set_(&GPIO.out_w1ts.val),
```